### ISSUE TITLE
Document changes for migration ([see example](https://github.com/carbon-design-system/ibm-products/pull/4139))

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4239]

**Comment by elycheea** [comment]
### Prop updates

This is as implemented in initial PR. Props still subject to change during subsequent reviews, especially around the time of release review.

e.g. `placeHolderText` may become `placeholderText`.


Old prop               | New prop               | Change      
-----------------------|------------------------|--------------
`className`            |`className`             | —
`clearButtonLabelText` | `clearButtonLabelText` | —
`hideScopesLabel`      | `hideScopesLabel`      | —
`labelText`            | `labelText`            | No longer required
`onChange`             | `onChange`             | —
`onSubmit`             | `onSubmit`             | —
`placeHolderText`      | `placeHolderText`      | —<br/> Suggest changing to `placeholderText`
`scopeToString`        | `scopeToString`        | —
`scopes`               | `scopes`               | —
`scopesTypeLabel`      | `scopesTypeLabel`      | —
`selectedScopes`       | `selectedScopes`       | —
`sortItems`            | `sortItems`            | —
`submitLabel`          | `submitLabel`          | —
`titleText`            | `titleText`            | —
`translateWithId`      | `translateWithId`      | —
`value`                | `value`                | —


Additional notes —
- FYI, noticing that several of the props, (`scopesTypeLabel`, `scopes`, `selectedScopes`, `translateWithId`, `sortItems`) are meant to be passed to the `MultiSelect`. Since these are all related, I wonder if it would make sense to follow the approach from @davidmenendez where we have `Dropdown` in `AddSelect` and we group these related props in an object/array.
- Should set controls for more search bar props in Storybook.

**Comment by davidmenendez** [comment]
> FYI, noticing that several of the props, (scopesTypeLabel, scopes, selectedScopes, translateWithId, sortItems) are meant to be passed to the MultiSelect. Since these are all related, I wonder if it would make sense to follow the approach from @davidmenendez where we have Dropdown in AddSelect and we group these related props in an object/array.

yeah i agree with this. basically any time that we're working with multiple carbon components we should do our best to enable users to pass as many props to those components for maximum flexibility. unless there's a specific callout in the design to not use something specific we should probably have something like `btnProps = {}` and `multiSelectProps = {}`
